modules/surrealdb/surrealdb.go

```diff
@@ -0,0 +1,117 @@
+package surrealdb
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// SurrealDBContainer represents the SurrealDB container type used in the module
+type SurrealDBContainer struct {
+	testcontainers.Container
+}
+
+// ConnectionString returns the connection string for the OpenLDAP container
+func (c *SurrealDBContainer) URL(ctx context.Context) (string, error) {
+	containerPort, err := c.MappedPort(ctx, "8000/tcp")
+	if err != nil {
+		return "", err
+	}
+
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	connStr := fmt.Sprintf("ws://%s/rpc", net.JoinHostPort(host, containerPort.Port()))
+	return connStr, nil
+}
+
+// WithUser sets the initial username to be created when the container starts
+// It is used in conjunction with WithPassword to set a username and its password.
+// It will create the specified user with superuser power.
+func WithUsername(username string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["SURREAL_USER"] = username
+	}
+}
+
+// WithPassword sets the initial password of the user to be created when the container starts
+// It is used in conjunction with WithUsername to set a username and its password.
+// It will set the superuser password for MongoDB.
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["SURREAL_PASS"] = password
+	}
+}
+
+// WithAuthentication enables or disables authentication for the SurrealDB instance
+func WithAuthentication(enabled bool) testcontainers.CustomizeRequestOption {
```

Comment:
If the default is auth=true, then I'd create an option only for disabling it, something like `WithoutAuthentication` or `NoAuthentication`.

Thoughts?